otherlibs/stdune-unstable/fpath.mli

```diff
@@ -1,11 +1,18 @@
 (** Functions on paths that are represented as strings *)
 
-(** Represent the result of [mkdir_p] *)
-type mkdir_p =
-  | Already_exists  (** The path already exists. No action was taken *)
+type mkdir_result =
+  | Already_exists  (** The directory already exists. No action was taken. *)
   | Created  (** The directory was created. *)
+  | Missing_parent_directory
```

Comment:
Perhaps slightly cleaner would be to make this:

```
type mkdir_p_result = Mkdir_p of mkdir_result | Missing_parent_directory
```

Reply:
Yeah, I was hesitating between these two variants. Call sites looked a little nicer with my variant, so I went for it. I don't mind if it's changed though.